### PR TITLE
add gocritic linter

### PR DESCRIPTION
**Pull request by jochil**
and improve existing linter setup by using `golangci-lint` instead of managing single linter

### CODE REVIEW

File: internal/cmd/init/init_test.go [hunk 1]
```diff
@@ -16,15 +15,15 @@ var baseTempDir string
...
+	baseTempDir, err = os.MkdirTemp("", "init-cmd-test-")
 	if err != nil {
 		log.Fatalf("Failed to create temp dir for tests: %+v", err)
 	}
 	defer fileutil.Cleanup(baseTempDir)
 
 	err = os.Chdir(baseTempDir)
 	if err != nil {
-		log.Fatalf("Failed to change the working directory to %s", baseTempDir)
+		log.Fatalf("Failed to change the working directory to %s", baseTempDir) //nolint:gocritic
```
Comment (fmeum) [inline internal/cmd/init/init_test.go:26]: If this is about defer not running, shouldn't we invoke the cleanup manually?

File: internal/cmd/root/root.go [hunk 2]
```diff
@@ -67,11 +67,11 @@ func New() *cobra.Command {
 
 	rootCmd.PersistentFlags().BoolP("verbose", "v", false,
 		"Show more verbose output, can be helpful for debugging problems")
-	viper.BindPFlag("verbose", rootCmd.PersistentFlags().Lookup("verbose"))
+	viper.BindPFlag("verbose", rootCmd.PersistentFlags().Lookup("verbose")) //nolint: errcheck
```
Comment (fmeum) [inline internal/cmd/root/root.go:70]: A brief comment explaining why we need to disable this check would be great here.

**Review by fmeum — APPROVED** [review]

File: internal/cmd/init/init_test.go (hunk 1 above)
Comment (jochil) [reply internal/cmd/init/init_test.go]: of course... changed it

File: internal/cmd/root/root.go (hunk 2 above)
Comment (jochil) [reply internal/cmd/root/root.go]: I just added the error handling ;)

**Pull request merged**